x/valset-pref/validator_set.go

```diff
@@ -228,33 +228,33 @@ func (k Keeper) UndelegateFromValidatorSet(ctx sdk.Context, delegatorAddr string
 	return nil
 }
 
-func (k Keeper) getValsetRatios(ctx sdk.Context, delegator sdk.AccAddress,
-	prefs []types.ValidatorPreference, undelegateAmt sdk.Int) ([]ValRatio, map[string]stakingtypes.Validator, sdk.Dec, error) {
+func (k Keeper) getValsetRatios(ctx sdk.Context,
+	prefs []types.ValidatorPreference, delegations []stakingtypes.Delegation,
+	undelegateAmt sdk.Int) ([]ValRatio, map[string]stakingtypes.Validator, sdk.Dec, error) {
 	// total amount user has delegated
 	totalDelegatedAmt := sdk.ZeroDec()
 	var valSetRatio []ValRatio
 	validators := map[string]stakingtypes.Validator{}
-	for _, val := range prefs {
+	for i, val := range prefs {
+		delegation := delegations[i]
+		if delegation.Shares.IsZero() {
+			// TODO: Define what we do
```

Comment:
i dont fully understand this problem. If the delegation shares is 0, we return or skip to the next validator no? 

also how are we getting delegations with 0 shares in the list because if delegation.Shares is 0 this is what GetDelegation() returns  
```
delegator_address: ""
validator_address: ""
shares: <nil>
 ```

Reply:
ah your right, I misremembered: https://github.com/cosmos/cosmos-sdk/issues/10216